### PR TITLE
Work on the move command

### PR DESCRIPTION
**Pull request by Ricardo-Osorio**
Overhaul many of the user facing messages to improve readability and visibility of what's happening during the process. 
Now also starts by reporting what type of scenario was detected to be running, I.E. Moving master, Moving replica or Promoting master.

Improved code comments in some places.

Added more information to the command description.

Removed the node IDs from messages due to not adding much. These are internal IDs known to CP, not k8s IDs the user can refer to.

Changed replica sync message to format the bytes into a human readable unit first (i.e. GiB, MiB, etc).
These updates now happen every 8sec (+3 sec) as these were happening slightly too often.

Force a timeout of 30 minutes when the existing one is smaller than that value. This happens only when a new replica has to be created.

Removed unused variable (timeout) from struct that holds all the vars for the move cmd.

Haven't managed to address the issue of not being able to control which replica gets removed at the end of the process so that part still works the same as before.
Neither `Evict()` or `SetPreferedForEviction()` helps in this case due to background threads in the CP adjusting to the changes we make from the CLI before we can configure the volume as we desire (at least reliably).

An example of the output of the command:
```bash
./bin/storageos update volume move pvc-46026b69-0a4a-45af-9fd5-8e9c08fb2bb7 ricardo-md-0-cw7cw ricardo-md-0-8frsm
Detected: moving master
Creating a new replica on [destination] node.

The duration of syncing operations depend on the rate of data transfers and network bandwidth.
A default timeout of 30min is in place but it should be adjusted with the global --timeout flag to adapt to both volume and environment.

Replica syncing...
	There are 15 GiB of data left to sync
	There are 14 GiB of data left to sync
	There are 13 GiB of data left to sync
	There are 12 GiB of data left to sync
	There are 10 GiB of data left to sync
	There are 9.8 GiB of data left to sync
	There are 8.2 GiB of data left to sync
	There are 7.4 GiB of data left to sync
	There are 6.0 GiB of data left to sync
	There are 5.1 GiB of data left to sync
	There are 4.2 GiB of data left to sync
	There are 2.7 GiB of data left to sync
	There are 2.0 GiB of data left to sync
	There are 553 MiB of data left to sync
Replica on [destination] node fully synced
Replica on [destination] node attempting promotion to master
Replica on [destination] node has now become the master
```

### CODE REVIEW

File: cmd/update/volume_move.go [hunk 1]
```diff
@@ -108,14 +105,43 @@ func (c *volumeMoveCommand) runWithCtx(ctx context.Context, cmd *cobra.Command,
...
+	// happen back to the user in a user-friendly msg
+	if isMovingMaster {
+		if hasReplicaOnTargetNode {
+			c.display.TextMessage(ctx, c.writer, "Detected: promoting replica")
+		} else {
+			c.display.TextMessage(ctx, c.writer, "Detected: moving master")
+		}
+	} else {
+		c.display.TextMessage(ctx, c.writer, "Detected: moving replica")
+	}
```
Comment (Mojachieee) [inline cmd/update/volume_move.go:122]: `Detected` might be a bit confusing for the user.`Performing action` could be better

File: cmd/update/volume_move.go [hunk 2]
```diff
@@ -108,14 +105,43 @@ func (c *volumeMoveCommand) runWithCtx(ctx context.Context, cmd *cobra.Command,
...
+The duration of syncing operations depend on the rate of data transfers and network bandwidth.
+A default timeout of 30min is in place but it should be adjusted with the global --timeout flag to adapt to both volume and environment.
+`)
+
+		// when we need to sync a new replica, force the use of a timeout with
+		// a minimum of 30 minutes (if not already set) to avoid unecessary
+		// errors due to not using a long enough timeout.
+		//
+		// most users will be moving pretty large volumes of data anyways.
+		if deadline, _ := ctx.Deadline(); time.Until(deadline) < time.Minute*30 {
```
Comment (Mojachieee) [inline cmd/update/volume_move.go:139]: How does this work if no deadline is set? 

Working it through in my head: `deadline` will be the zero time value, but then what does `time.Until(deadline)` return?

File: cmd/update/volume_move.go [hunk 3]
```diff
@@ -133,17 +159,21 @@ func (c *volumeMoveCommand) runWithCtx(ctx context.Context, cmd *cobra.Command,
...
-				c.display.TextMessage(ctx, c.writer, "failed to get updated volume config")
+				c.display.TextMessage(ctx, c.writer, fmt.Sprintf("Failed to restore original replica count: failed to get updated volume config: %s", err.Error()))
 				return
 			}
 
 			currentReplicaCount := len(updatedVol.Replicas)
 
 			if currentReplicaCount-originalReplicaCount != 1 {
-				c.display.TextMessage(ctx, c.writer, "replica count of volume has been changed outside of move operation")
+				c.display.TextMessage(ctx, c.writer, "Replica count has been changed outside of move operation, aborting automatic deletion. Volume replicas needs to be managed manually.")
```
Comment (Mojachieee) [inline cmd/update/volume_move.go:176]: What does this involve? Can we tell users what to do as part of this message or should we have some knowledge base article about it?

**Review by Mojachieee — COMMENTED** [review]
Left a few questions, that should be answered. Apart from then it looks good. 

We'll need to re-do an operator release once we've released this. As the CLI is now bundle into the operator

File: cmd/update/volume_move.go (hunk 3 above)
Comment (Ricardo-Osorio) [reply cmd/update/volume_move.go:176]: The user will read something like this (and in this order):

"Creating replica on [destination] node"
"Syncing replica..."
"Replica count has been changed outside of move operation, aborting automatic deletion. Volume replicas needs to be managed manually." - This message

So I hope either the user:
- understands the last message means the first step was not reverted
- reads the volume replicas need to be managed somehow and then checking the replicas understand what happened and adjusts the replica count as he needs

File: cmd/update/volume_move.go (hunk 2 above)
Comment (Ricardo-Osorio) [reply cmd/update/volume_move.go:139]: There's a default one in place (15s I believe)

File: cmd/update/volume_move.go (hunk 3 above)
Comment (Mojachieee) [reply cmd/update/volume_move.go:176]: Sorry, I should've been more specific. I meant what does `Volume replicas needs to be managed manually.` involve. Telling the user to select the correct replica and adjust the replica count is probably too much for a log message 😓

**Review by Mojachieee — APPROVED** [review]

**Pull request merged**